ladybug/src/main/java/nl/nn/ibistesttool/filter/BlackBox.java

```diff
@@ -15,55 +15,53 @@
-import nl.nn.testtool.Report;
-import nl.nn.testtool.filter.CheckpointMatcher;
 
 /**
+ * Only show senders that communicate with another system
+ * 
  * @author Jaco de Groot
  */
-public class BlackBox implements CheckpointMatcher {
-	
-	public boolean match(Report report, Checkpoint checkpoint) {
-		if (
-				checkpoint.getName() != null
-				&&
-				(
-					checkpoint.getName().startsWith("Sender ")
-					// Also in stub4testtool.xsl
-					&& !"nl.nn.adapterframework.jdbc.DirectQuerySender".equals(checkpoint.getSourceClassName())
-					&& !"nl.nn.adapterframework.jdbc.FixedQuerySender".equals(checkpoint.getSourceClassName())
-					&& !"nl.nn.adapterframework.senders.DelaySender".equals(checkpoint.getSourceClassName())
-					&& !"nl.nn.adapterframework.senders.EchoSender".equals(checkpoint.getSourceClassName())
-					&& !"nl.nn.adapterframework.senders.IbisLocalSender".equals(checkpoint.getSourceClassName())
-					&& !"nl.nn.adapterframework.senders.LogSender".equals(checkpoint.getSourceClassName())
-					&& !"nl.nn.adapterframework.senders.ParallelSenders".equals(checkpoint.getSourceClassName())
-					&& !"nl.nn.adapterframework.senders.SenderSeries".equals(checkpoint.getSourceClassName())
-					&& !"nl.nn.adapterframework.senders.SenderWrapper".equals(checkpoint.getSourceClassName())
-					&& !"nl.nn.adapterframework.senders.XsltSender".equals(checkpoint.getSourceClassName())
-					// Not in stub4testtool.xsl
-					&& !"nl.nn.adapterframework.senders.FixedResultSender".equals(checkpoint.getSourceClassName())
-					&& !"nl.nn.adapterframework.senders.XmlValidatorSender".equals(checkpoint.getSourceClassName())
-				)
-			) {
+public class BlackBox extends GrayBox {
+	private Set<String> SENDERS_TO_HIDE = new HashSet<>(Arrays.asList(
+			// Also in stub4testtool.xsl
+			"nl.nn.adapterframework.jdbc.ResultSet2FileSender",
+			"nl.nn.adapterframework.jdbc.DirectQuerySender",
+			"nl.nn.adapterframework.jdbc.FixedQuerySender",
+			"nl.nn.adapterframework.jdbc.XmlQuerySender",
+			"nl.nn.adapterframework.senders.DelaySender",
+			"nl.nn.adapterframework.senders.EchoSender",
+			"nl.nn.adapterframework.senders.IbisLocalSender",
+			"nl.nn.adapterframework.senders.LogSender",
+			"nl.nn.adapterframework.senders.ParallelSenders",
+			"nl.nn.adapterframework.senders.SenderSeries",
+			"nl.nn.adapterframework.senders.SenderWrapper",
+			"nl.nn.adapterframework.senders.XsltSender",
+			"nl.nn.adapterframework.senders.FixedResultSender",
+			"nl.nn.adapterframework.senders.JavascriptSender",
+			"nl.nn.adapterframework.jdbc.MessageStoreSender",
+			"nl.nn.adapterframework.senders.ReloadSender",
+			"nl.nn.adapterframework.compression.ZipWriterSender",
+			"nl.nn.adapterframework.senders.LocalFileSystemSender",
+			// Not in stub4testtool.xsl
+			"nl.nn.adapterframework.senders.XmlValidatorSender"));
+
+	@Override
+	protected boolean isSender(Checkpoint checkpoint) {
+		if (checkpoint.getName() != null && checkpoint.getName().startsWith("Sender ")
```

Comment:
```suggestion
		return (checkpoint.getName() != null && checkpoint.getName().startsWith("Sender ") ...
```
Dit kan nog net iets korter 😄